driver-async/src/main/com/mongodb/async/client/OperationIterable.java

```diff
@@ -117,6 +117,12 @@ public void onResult(final List<T> results, final Throwable t) {
         return new MappingIterable<T, U>(this, mapper);
     }
 
+    @Override
+    public OperationIterable<T> batchSize(final int batchSize) {
```

Comment:
I don't think it's fine, because FindFluentImp.map(...) returns a MappingIterable, which is constructed with an OperationIterable, so when you call batchSize on the MappingIterable, it delegates that to the OperationIterable, which drops it.